dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -50,28 +52,31 @@ Block HashJoinProbeBlockInputStream::getTotals()
         }
         join->joinTotals(totals);
     }
+
     return totals;
 }
 
 Block HashJoinProbeBlockInputStream::getHeader() const
 {
     Block res = children.back()->getHeader();
-    join->joinBlock(res);
+    assert(res.rows() == 0);
+    join->joinBlock(res, *probe_process_info_ptr);
```

Comment:
I think we can construct a tmp `probe_process_info` instead of reuse the one which is used for probe?

Reply:
> I think we can construct a tmp `probe_process_info` instead of reuse the one which is used for probe?

done.